webui/react/src/components/ExperimentRetainLogsModal.tsx

```diff
@@ -0,0 +1,163 @@
+}: Props) => {
+  const idPrefix = useId();
+  const { openToast } = useToast();
+  const [checked, setChecked] = useState<boolean>(false);
+  const [form] = Form.useForm<FormInputs>();
+  const inputDays = Form.useWatch('numDays', form);
+
+  const handleCheckBoxChange = useCallback(
+    (event: CheckboxChangeEvent) => {
+      const isChecked = event.target.checked;
+      setChecked(isChecked);
+      if (isChecked) {
+        form.setFieldValue('numDays', FOREVER);
+      } else {
+        form.setFieldValue('numDays', null);
+      }
+    },
+    [form],
+  );
+
+  const handleSubmit = useCallback(async () => {
+    const values = await form.validateFields();
+    const numberDays = values.numDays;
+    let filt = filters;
+    if (excludedExperimentIds && excludedExperimentIds.size > 0) {
+      filt = { ...filters, excludedExperimentIds: Array.from(excludedExperimentIds) };
+    }
+    try {
+      const results = await changeExperimentLogRetention({
+        experimentIds,
+        filters: filt,
+        numDays: numberDays,
+      });
+
+      onSubmit?.(results.successful);
+
+      const numSuccesses = results.successful.length;
+      const numFailures = results.failed.length;
+
+      const stringDays = numberDays === -1 ? 'forever' : `for ${numberDays} days`;
+
+      if (numFailures === 0) {
+        openToast({
+          closeable: true,
+          description: `Retained logs for ${results.successful.length} ${pluralizer(
+            experimentIds.length,
+            'experiment',
+          )} ${stringDays}`,
+          title: 'Retain Logs Success',
+        });
+      } else if (numSuccesses === 0) {
+        openToast({
+          description: `Unable to retain logs for ${numFailures} ${pluralizer(
+            experimentIds.length,
+            'experiment',
+          )}`,
+          severity: 'Warning',
+          title: 'Retain Logs Failure',
+        });
+      } else {
+        openToast({
+          closeable: true,
+          description: `Failed to retain logs for ${numFailures} experiments out of ${
+            numFailures + numSuccesses
+          } for ${numberDays} days.`,
+          severity: 'Warning',
+          title: 'Partial Retain Logs Failure',
+        });
+      }
+    } catch (e) {
+      if (e instanceof DetError) {
+        handleError(e, {
+          level: e.level,
+          publicMessage: e.publicMessage,
+          publicSubject: 'Unable to retain logs.',
+          silent: false,
+          type: e.type,
+        });
+      } else {
+        handleError(e, {
+          level: ErrorLevel.Error,
+          publicMessage: 'Please try again later.',
+          publicSubject: 'Unable to aretain logs.',
+          silent: false,
+          type: ErrorType.Server,
+        });
+      }
+    }
+  }, [form, excludedExperimentIds, experimentIds, filters, onSubmit, openToast]);
+
+  return (
+    <Modal
+      cancel
+      size="small"
+      submit={{
+        disabled: inputDays == null,
```

Comment:
Isn't that because the submit button needs to be disabled if there is no input? Once we check the Forever checkbox, we have an input so it the submit button will get re-enabled

Reply:
confirmed its from hew lib.